lisa/analysis/idle.py

```diff
@@ -72,6 +73,50 @@ def signal_cpu_active(self, cpu):
         # Fix sequences of wakeup/sleep events reported with the same index
         return handle_duplicate_index(cpu_active)
 
+    @memoized
+    @signal_cpu_active.used_events
+    @TasksAnalysis.df_tasks_states.used_events
+    def signal_cpu_sched_active(self, cpu):
```

Comment:
Do we want to create a new function or just add an `as_sched=False` parameter to the existing `signal_cpu_active()` ?

Reply:
Oh, I just remembered why I did this: they have different events requirements. One only needs `cpu_idle`, the other needs `cpu_idle`, `sched_switch` and `sched_wakeup`. I don't expect anyone to trace `cpu_idle` without tracing `sched_switch`, but it's easy to forget `sched_wakeup`, so I kept these separate.

Reply:
+1, it's better to avoid function that seemingly depend on everything, or optionally depend on everything and then do manual validation. Even when using optional events, everything ends up being traced and event list from `lisa-test --list` would quickly become meaningless.